game/src/game/components/Controllers/AIController.cpp

```diff
@@ -129,27 +145,32 @@ void AIController::OnUpdate(const Timestep& delta_time)
     }
 }
 
-void AIController::FixRespawnOrientation(
-    const glm::vec3& next_checkpoint_location,
-    const glm::vec3& checkpoint_location)
+void AIController::FixRespawnOrientation(const glm::vec3& next_checkpoint,
+                                         const glm::vec3& last_checkpoint)
 {
-    transform_->SetOrientation(kIdentityQuat);
+    auto current_orientation = transform_->GetOrientation();
+
+    // assume car is initially oriented along the negative z-axis
+    glm::vec3 forward = transform_->GetForwardDirection();
+    glm::vec3 direction = glm::normalize(next_checkpoint - last_checkpoint);
+    glm::vec3 axis = glm::normalize(glm::cross(forward, direction));
+    float angle = glm::acos(glm::dot(forward, direction));
+
+    transform_->SetOrientation(glm::angleAxis(angle, axis) *
+                               current_orientation);
 }
 
 void AIController::HandleRespawn(const Timestep& delta_time)
 {
-    // TODO: if the AI skips a checkpoint by mistake, then wait 5 seconds, if it
-    //          doesnt correct itself then respawn it.
-
     // if the velocity of car is less than some amount then respawn the car.
-    // HandleMinSpeedThresholdRespawn(delta_time);
+    HandleMinSpeedThresholdRespawn(delta_time);
```

Comment:
Is this still needed after the changes Julian made for making a non-grounded car respawn?